src/sidebar/components/Annotation/AnnotationShareControl.tsx

```diff
@@ -62,8 +66,10 @@ function AnnotationShareControl({
   const toggleSharePanel = () => setOpen(!isOpen);
   const closePanel = () => setOpen(false);
 
-  // Interactions outside of the component when it is open should close it
-  useElementShouldClose(shareRef, isOpen, closePanel);
+  // Interactions outside the component when it is open should close it
+  useClickAway(shareRef, closePanel, { enabled: isOpen });
+  useFocusAway(shareRef, closePanel, { enabled: isOpen });
+  useKeyPress(['Escape'], closePanel, { enabled: isOpen });
```

Comment:
https://github.com/hypothesis/frontend-shared/issues/900 documented the rationale for decomposing the `useElementShouldClose` hook into separate hooks, but I think it would be useful to have a combined hook that captured the set of actions that close popover-type controls.

Reply:
Yep, maybe you are right. Perhaps what we should do is keep `useElementShouldClose`, un-deprecate it, and just make it a wrapper of the other three hooks.

What do you think?

Reply:
I think I would add a new `usePopoverShouldClose` hook that wraps the existing hooks and is more specific about the kind of widget it works with. The problem with `useElementShouldClose` was that we used it for popover-type controls but also different controls that needed different close behaviors.

Reply:
Perfect. I'll close this one for now